packages/react-components/theme-designer/src/components/Nav/Nav.tsx

```diff
@@ -32,17 +34,20 @@ const useStyles = makeStyles({
   },
 });
 
-export const Name = () => {
-  const styles = useStyles();
-  return (
-    <div className={styles.element}>
-      <Text>Untitled</Text>
-    </div>
-  );
-};
-
 export const Nav: React.FC<NavProps> = props => {
   const styles = useStyles();
+
+  const underlineId = useId('input-underline');
+  const name = useContextSelector(AppContext, ctx => ctx.name);
+  const setName = useContextSelector(AppContext, ctx => ctx.setName);
+
+  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newName = e.target.value;
+    if (newName.match('^[a-zA-Z0-9]*$')) {
```

Comment:
don't restrict the name and force the user to know string options, parse it for them